src/figuresClasses.ts

```diff
@@ -1,19 +1,100 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
 
-}
+  sides: number[];
+
+  color: 'red' | 'green' | 'blue';
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.sides = [a, b, c];
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All properties should be positive!');
+    }
 
-export class Circle {
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error(`Triangle with sides ${a}, ${b}, and ${c} can't exist`);
+    }
+  }
 
+  getArea(): number {
+    const [a, b, c] = [...this.sides];
+    const perimeter = (a + b + c);
+
```

Comment:
Instead of calculating `perimeter / 2` multiple times, it's better to store it in a variable.